packages/toast/src/toast.ts

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    html,
+    CSSResultArray,
+    TemplateResult,
+    LitElement,
+    property,
+    PropertyValues,
+} from 'lit-element';
+import '@spectrum-web-components/button';
+import '@spectrum-web-components/icon';
+import '@spectrum-web-components/icons';
+
+import toastStyles from './toast.css.js';
+
+export const toastVariants = [
+    'negative',
+    'positive',
+    'info',
+    'error',
+    'warning',
+];
+
+/**
+ * @slot - The toast content
+ */
+
+export class Toast extends LitElement {
+    public static get styles(): CSSResultArray {
+        return [toastStyles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: Boolean, reflect: true })
+    public disappearing = false;
```

Comment:
not sure if this is the responsibility of swc to enforce, but looking at the toast design on spectrum.adobe.com, anything that is "actionable" is should not be able to be disappearing:
https://spectrum.adobe.com/page/toast/#Priority-queue

Reply:
I'd vote for `auto-dismiss` rather than `disappearing` for the naming here.

Reply:
@davidyxu It's for this reason that I added this as an alternative instead of the default. For that, I am working from https://github.com/adobe/react-spectrum-v3/blob/master/specs/accessibility/Toast.mdx as supplied by @jnurthen  as the source of the timing decisions here. I should have documented it some way, but I meant to deliver a baseline, and then extend functionality later to manage text with more than 120 words. Maybe it makes sense to surface that math as an exported function? A la:
```
export const howLongToWait = (words: number | string) => number; // equalling 1 for every 120 words
```

In that case, @adixon-adobe rather than relying on `disappearing` or `auto-dismiss` we can match more closely to https://github.com/adobe/react-spectrum-v3/blob/master/packages/%40react-types/toast/src/index.d.ts#L9 and use `timeout` to signify that a toast is self-dismissing?

Reply:
The timeout option used in react-spectrum makes the most sense to me, where the lack of a timeout signifies that it isn't self-dismissing. I guess the downside to this approach is that there wouldn't be an easy way to provide a sensible default timeout setting?

Reply:
I think _maybe_ we could still require the timeout to be at least 6000 milliseconds, as it’s the baseline a11y requirement. But, that’s a great point!

Reply:
I think in an ideal world, we could raise console warnings when we go below a recommended threshold when in dev mode, but these get compiled out when building for prod. I'm not aware of any clean way of setting this generically in SWC.